docs/source/common/production_inference.rst

```diff
@@ -1,48 +1,252 @@
 .. _production_inference:
 
+#######################
 Inference in Production
-=======================
-PyTorch Lightning eases the process of deploying models into production.
+#######################
 
+Once a model is trained, deploying to production and running inference is the next task. To help you with it, following are the possible approaches
+you can use to deploy and use your models for inference.
 
-Exporting to ONNX
------------------
-PyTorch Lightning provides a handy function to quickly export your model to ONNX format, which allows the model to be independent of PyTorch and run on an ONNX Runtime.
+------------
 
-To export your model to ONNX format call the ``to_onnx`` function on your Lightning Module with the filepath and input_sample.
+******************
+With Lightning API
+******************
+
+Following are some possible ways you can use Lightning to run inference in production. Note that this can introduce extra required Lightning dependencies
+in your production environment.
+
+------------
+
+Prediction API
+==============
+
+Lightning provides you with a prediction API that can be accessed using :meth:`~pytorch_lightning.trainer.trainer.Trainer.predict`.
+To configure this with LightningModule, you need to override the :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` method.
+By default :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` calls the :meth:`~pytorch_lightning.core.lightning.LightningModule.forward`
+method. In order to customize this behaviour, simply override the :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` method.
+
+For the example let's override ``predict_step`` and try out `Monte Carlo Dropout <https://arxiv.org/pdf/1506.02142.pdf>`_:
 
 .. code-block:: python
 
-    filepath = "model.onnx"
+    class LitMCdropoutModel(pl.LightningModule):
+        def __init__(self, model, mc_iteration):
+            super().__init__()
+            self.model = model
+            self.dropout = nn.Dropout()
+            self.mc_iteration = mc_iteration
+
+        def predict_step(self, batch, batch_idx):
+            # enable Monte Carlo Dropout
+            self.dropout.train()
+
+            # take average of `self.mc_iteration` iterations
+            pred = [self.dropout(self.model(x)).unsqueeze(0) for _ in range(self.mc_iteration)]
+            pred = torch.vstack(pred).mean(dim=0)
+            return pred
+
+------------
+
+PyTorch Runtime
+===============
+
+You can also load the saved checkpoint and use it as a regular :class:`torch.nn.Module`.
```

Comment:
```suggestion
It is possible to use your checkpoint with regular :class:`torch.nn.Module`.
```

Reply:
@rohitgr7 This example is confusing as the SimpleModel is actually a LightningModule.

Reply:
yes, but this is under `Within Lightning` section, and since it's just an nn.Module, here we are telling that LightningModule instances are not different and can be used just like an `nn.Module` instance by just calling `model(inputs)`.